PublicMapping/districtbuilder
# Add user account page

## Overview

This PR:
- updates the `InputField` component to take more optional arguments
- adds a dropdown menu item `Account` to the avatar component in the app header
- updates the data model of the user `PATCH` so that it takes `isMarketingEmailOn` field
- adds a user account page
- updates the forgot password page so that it reads an optional route param to populate the email field

### Checklist


### Demo

![Screen Shot 2022-06-14 at 2 14 36 PM](https://user-images.githubusercontent.com/16109558/173660373-c80d92a0-1e4e-4034-a44b-911e0ffc0cf9.png)


## Testing Instructions

- Run update script to install a new dependency
- Spin up local server and client
- Log in to the app
- Click the avatar from app topbar and select the `Account` menu item
- On the user account page, make sure all the fields are populated with existing info and email field cannot be edited
- Make sure it allows you to submit the form if you have changed the name field and/or the marketing email checkbox
- Make sure the data is persisted to the DB and the page fetches the new user account data after successful submission
- Make sure clicking on the `Reset password` link takes you to the forgot password page with your email pre-filled
- Go back to the user account page. Make sure the email link also works under the account deletion section

Closes #1169
